Source/AwsCommonRuntimeKit/crt/CRTErrorGenerated.swift

```diff
@@ -0,0 +1,307 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+// This file is generated using Script/CRTErrorGenerator.swift.
+// Do not modify this file.
+
+import AwsCCommon
+
+/// Error type for CRT errors thrown from C code
+public enum CRTError: Int32, Error {
+
+    case UNKNOWN_ERROR_CODE = -1
+
+    /// AWS-C-COMMON
+    case AWS_ERROR_SUCCESS = 0
+    case AWS_ERROR_OOM = 1
+    case AWS_ERROR_NO_SPACE = 2
+    case AWS_ERROR_UNKNOWN = 3
+    case AWS_ERROR_SHORT_BUFFER = 4
+    case AWS_ERROR_OVERFLOW_DETECTED = 5
+    case AWS_ERROR_UNSUPPORTED_OPERATION = 6
+    case AWS_ERROR_INVALID_BUFFER_SIZE = 7
+    case AWS_ERROR_INVALID_HEX_STR = 8
+    case AWS_ERROR_INVALID_BASE64_STR = 9
+    case AWS_ERROR_INVALID_INDEX = 10
+    case AWS_ERROR_THREAD_INVALID_SETTINGS = 11
+    case AWS_ERROR_THREAD_INSUFFICIENT_RESOURCE = 12
+    case AWS_ERROR_THREAD_NO_PERMISSIONS = 13
+    case AWS_ERROR_THREAD_NOT_JOINABLE = 14
+    case AWS_ERROR_THREAD_NO_SUCH_THREAD_ID = 15
+    case AWS_ERROR_THREAD_DEADLOCK_DETECTED = 16
+    case AWS_ERROR_MUTEX_NOT_INIT = 17
+    case AWS_ERROR_MUTEX_TIMEOUT = 18
+    case AWS_ERROR_MUTEX_CALLER_NOT_OWNER = 19
+    case AWS_ERROR_MUTEX_FAILED = 20
+    case AWS_ERROR_COND_VARIABLE_INIT_FAILED = 21
+    case AWS_ERROR_COND_VARIABLE_TIMED_OUT = 22
+    case AWS_ERROR_COND_VARIABLE_ERROR_UNKNOWN = 23
+    case AWS_ERROR_CLOCK_FAILURE = 24
+    case AWS_ERROR_LIST_EMPTY = 25
+    case AWS_ERROR_DEST_COPY_TOO_SMALL = 26
+    case AWS_ERROR_LIST_EXCEEDS_MAX_SIZE = 27
+    case AWS_ERROR_LIST_STATIC_MODE_CANT_SHRINK = 28
+    case AWS_ERROR_PRIORITY_QUEUE_FULL = 29
+    case AWS_ERROR_PRIORITY_QUEUE_EMPTY = 30
+    case AWS_ERROR_PRIORITY_QUEUE_BAD_NODE = 31
+    case AWS_ERROR_HASHTBL_ITEM_NOT_FOUND = 32
+    case AWS_ERROR_INVALID_DATE_STR = 33
+    case AWS_ERROR_INVALID_ARGUMENT = 34
+    case AWS_ERROR_RANDOM_GEN_FAILED = 35
+    case AWS_ERROR_MALFORMED_INPUT_STRING = 36
+    case AWS_ERROR_UNIMPLEMENTED = 37
+    case AWS_ERROR_INVALID_STATE = 38
+    case AWS_ERROR_ENVIRONMENT_GET = 39
+    case AWS_ERROR_ENVIRONMENT_SET = 40
+    case AWS_ERROR_ENVIRONMENT_UNSET = 41
+    case AWS_ERROR_STREAM_UNSEEKABLE = 42
+    case AWS_ERROR_NO_PERMISSION = 43
+    case AWS_ERROR_FILE_INVALID_PATH = 44
+    case AWS_ERROR_MAX_FDS_EXCEEDED = 45
+    case AWS_ERROR_SYS_CALL_FAILURE = 46
+    case AWS_ERROR_C_STRING_BUFFER_NOT_NULL_TERMINATED = 47
+    case AWS_ERROR_STRING_MATCH_NOT_FOUND = 48
+    case AWS_ERROR_DIVIDE_BY_ZERO = 49
+    case AWS_ERROR_INVALID_FILE_HANDLE = 50
+    case AWS_ERROR_OPERATION_INTERUPTED = 51
+    case AWS_ERROR_DIRECTORY_NOT_EMPTY = 52
+    case AWS_ERROR_PLATFORM_NOT_SUPPORTED = 53
```

Comment:
I'd love to do a transform on these names to make them more Swift-like
ex:
```
AWS_ERROR_PLATFORM_NOT_SUPPORTED -> platformNotSupported
```

there would need to be some special cases since we weren't 100% uniform in how errors were named

Reply:
See: https://www.swift.org/documentation/api-design-guidelines/#follow-case-conventions

Reply:
Thanks, updated.